ArmaForces.Boderator/ArmaForces.Boderator.BotService/Discord/DiscordService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using ArmaForces.Boderator.BotService.DTOs;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace ArmaForces.Boderator.BotService.Discord
+{
+    public sealed class DiscordService : IDiscordService, IHostedService
+    {
+        private readonly ILogger<DiscordService> _log;
+        private readonly DiscordSocketClient _discordClient;
+        private string _token;
+
+        public DiscordService(ILogger<DiscordService> logger, string token)
+        {
+            _log = logger;
+            _discordClient = new DiscordSocketClient();
+            _discordClient.Log += message => Task.Run(() => _log.Log(MapSeverity(message.Severity), "[Discord.NET Log] " + message.Message));
+            _discordClient.Connected += () => Task.Run(() => _log.LogInformation("Discord connected"));
+            _token = token;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _log.LogInformation("Discord Service started");
+            await _discordClient.LoginAsync(TokenType.Bot, _token, true);
+            await _discordClient.StartAsync();
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken) => Task.Run(() =>
+        {
+            _discordClient.Dispose();
+            _log.LogInformation("Discord Service stopped");
+        }, cancellationToken);
+
+        public DiscordServiceStatus GetDiscordClientStatus()
+        {
+            _log.LogInformation($"Current Discord Bot status: Login: {_discordClient.LoginState} | " +
+                                $"Connection: {_discordClient.ConnectionState} | " +
+                                $"Status: {_discordClient.Status}");
+            return new DiscordServiceStatus
+            {
+                ConnectionState = _discordClient.ConnectionState.ToString(),
+                LoginState = _discordClient.LoginState.ToString(),
+                ClientState = _discordClient.Status.ToString()
```

Comment:
I think we can change these properties to enums instead of making strings. When we want a string in the API, we can change the enum serialization policy.

Reply:
We can use enum vals from Discord.Net lib

Reply:
That's what I meant of course. 👍🏻